ui/de-lib/src/main/java/org/iplantc/de/analysis/client/presenter/AnalysesPresenterImpl.java

```diff
@@ -286,33 +286,40 @@ public void go(final HasOneWidget container, final List<Analysis> selectedAnalys
     @Override
     public void loadAnalyses(AnalysisFilter filter) {
         FilterPagingLoadConfig config = loader.getLastLoadConfig();
-
         FilterConfigBean idParentFilter = new FilterConfigBean();
-        idParentFilter.setField(AnalysisSearchField.PARENT_ID);
-        idParentFilter.setValue("");
-
         FilterConfigBean filterCb = new FilterConfigBean();
         config.getFilters().clear();
-        switch (filter) {
-            case ALL:
-                filterCb.setField("ownership");
-                filterCb.setValue("all");
-                break;
-            case SHARED_WITH_ME:
-                filterCb.setField("ownership");
-                filterCb.setValue("theirs");
-                break;
-
-            case MY_ANALYSES:
-                filterCb.setField("ownership");
-                filterCb.setValue("mine");
-                break;
+        if (filter != null) {
+            idParentFilter.setField(AnalysisSearchField.PARENT_ID);
+            idParentFilter.setValue("");
+            switch (filter) {
+                case ALL:
+                    filterCb.setField("ownership");
+                    filterCb.setValue("all");
+                    break;
+                case SHARED_WITH_ME:
+                    filterCb.setField("ownership");
+                    filterCb.setValue("theirs");
+                    break;
+
+                case MY_ANALYSES:
+                    filterCb.setField("ownership");
+                    filterCb.setValue("mine");
+                    break;
+            }
+            config.getFilters().add(idParentFilter);
+            config.getFilters().add(filterCb);
+            config.setLimit(200);
+            config.setOffset(0);
+            loader.load(config);
+        } else {
+            idParentFilter.setField(AnalysisSearchField.PARENT_ID);
+            idParentFilter.setValue(view.getParentAnalysisId());
+            config.getFilters().add(idParentFilter);
+            config.setLimit(200);
+            config.setOffset(0);
+            loader.load(config);
```

Comment:
It looks like most of these are duplicated in both the `if` and `else` branches.
I think most of these can be factored outside the if/else block: the `config.getFilters().add(filterCb)` and `idParentFilter.setValue(...)` calls can stay inside and I think everything else can come after.